controllers/user_controller_test.go

```diff
@@ -154,6 +156,53 @@ var _ = Describe("UserController", func() {
 					})))
 			})
 		})
+
+		When("the user has limits defined", func() {
+			BeforeEach(func() {
+				userName = "test-user-limits"
+				userLimits = topology.UserLimits{
+					Connections: 5,
+					Channels:    10,
+				}
+				fakeRabbitMQClient.PutUserReturns(&http.Response{
+					Status:     "201 Created",
+					StatusCode: http.StatusCreated,
+				}, nil)
+				fakeRabbitMQClient.PutUserLimitsReturns(&http.Response{
+					Status:     "201 Created",
+					StatusCode: http.StatusCreated,
+				}, nil)
+			})
+
+			It("should create the user limits", func() {
+				Expect(k8sClient.Create(ctx, &user)).To(Succeed())
+				Eventually(func() []topology.Condition {
+					_ = k8sClient.Get(
+						ctx,
+						types.NamespacedName{Name: user.Name, Namespace: user.Namespace},
+						&user,
+					)
+
+					return user.Status.Conditions
+				}).
+					Within(statusEventsUpdateTimeout).
+					WithPolling(time.Second).
+					Should(ContainElement(MatchFields(IgnoreExtras, Fields{
+						"Type":   Equal(topology.ConditionType("Ready")),
+						"Reason": Equal("SuccessfulCreateOrUpdate"),
+						"Status": Equal(corev1.ConditionTrue),
+					})))
+				By("calling PutUserLimits with the correct user limits")
+				Expect(fakeRabbitMQClient.PutUserLimitsCallCount()).To(BeNumerically(">", 0))
+				_, userLimitsValues := fakeRabbitMQClient.PutUserLimitsArgsForCall(0)
+				connectionLimit, ok := userLimitsValues["max-connections"]
+				Expect(ok).To(BeTrue())
+				Expect(connectionLimit).To(Equal(5))
+				channelLimit, ok := userLimitsValues["max-channels"]
+				Expect(ok).To(BeTrue())
+				Expect(channelLimit).To(Equal(10))
```

Comment:
You can replace these lines with matches `ToHaveKeyWithValue()`. It's a bit easier to understand what's going, and in the event of a failure, you get something better than `expected false to be true` and `123 not equal 5`
https://onsi.github.io/gomega/#havekeywithvaluekey-any-value-any